pkg/config/config.go

```diff
@@ -123,19 +137,38 @@ func (r *KCLRun) Transform(rl *fn.ResourceList) error {
 		if err != nil {
 			return err
 		}
-		transformedObjects = append(transformedObjects, obj)
+
+		// Check if the transformed object matches the resource rules
+		if r.MatchResourceRules(obj) {
+			transformedObjects = append(transformedObjects, obj)
+		}
 	}
 	rl.Items = transformedObjects
 	return nil
 }
 
+// MatchResourceRules checks if the given Kubernetes object matches the resource rules specified in KCLRun.
+func (r *KCLRun) MatchResourceRules(obj *fn.KubeObject) bool {
+	// Iterate through each resource rule
+	for _, rule := range r.Spec.MatchConstraints.ResourceRules {
+		if containsString(rule.APIGroups, obj.GroupKind().Group) &&
+			containsString(rule.APIVersions, obj.GetAPIVersion()) &&
+			containsString(rule.Resources, obj.GetKind()) {
+			return true
+		}
+	}
+	return false
```

Comment:
When we do not set the MatchConstraints.ResourceRules field, I think it should return true?